### PR TITLE
Check for missing node attributes, warn and skip setting up client keys ...

### PR DESCRIPTION
**Pull request by AlexClineBB**
...on bad nodes.

### CODE REVIEW

File: recipes/wui.rb [hunk 1]
```diff
@@ -37,8 +37,8 @@
   members node['apache']['group']
 end
 
-apache_dir = node['apache']['dir']
-apache_doc_root = "#{apache_dir}/htdocs"
+apache_dir = node['apache']['docroot_dir']
+apache_doc_root = "#{apache_dir}"
```
Comment (jtimberman) [inline recipes/wui.rb]: Why is this change here?

File: recipes/server.rb [hunk 2]
```diff
@@ -34,6 +34,11 @@
 
   ssh_hosts << n['ipaddress'] if n['keys']
 
+  if(!n['fqdn'] || !n['ipaddress'] || !n['keys'])
+    Chef::Log.warn("Not setting up client keys for `#{n}` which is missing required information.")
+    next
+  end
```
Comment (jtimberman) [inline recipes/server.rb]: I'm not wild about this syntax. It seems really unclear and too much like "magic."

I'd rather have a helper library that checks for whatever is required at the top of the recipe and then bails out early if the requirements aren't met.

File: metadata.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@
 license          "Apache 2.0"
 description      "Installs/Configures ossec"
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version          "1.0.5"
+version          "1.0.6"
```
Comment (jtimberman) [inline metadata.rb]: Please don't increment versions in PR's. The maintainers will do that when they cut a new release.

File: Berksfile [hunk 4]
```diff
@@ -1,2 +1,3 @@
-site :opscode
+source "http://berkshelf.in.bbhosted.com"
```
Comment (jtimberman) [inline Berksfile]: You want:

```ruby
source 'https://supermarket.chef.io'
```

**Comment by tas50** [comment]
Closing this out since we don't do any of the WUI setup
